### PR TITLE
[Custom Fields] Align with FluxC storage changes - part 1

### PR DESCRIPTION
**Pull request by hichamboushaba**
Part of: #12212 

Depends on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3070, not ready for merge.

### Description
This PR just updates the domain code to align with the FluxC changes done on https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3070

### Testing information
- Smoke test the custom fields viewer in the order details, and look mainly for:
    - All the displayable (doesn't start with `_`) fields are available
    - No Json values are shown.
    - HTML tags are removed before showing HTML fields.
- Create an order that include subscription products, and confirm the order details in the app contain the data about the subscription.
- Test issuing a refund for an order, and confirm it works as expected

### Images/gif

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce-Wear/src/main/java/com/woocommerce/android/wear/model/Refund.kt [hunk 1]
```diff
@@ -55,7 +55,7 @@ fun WCRefundModel.WCRefundItem.toAppModel(): Refund.Item {
         -totalTax, // WCRefundItem.totalTax is NEGATIVE
         sku ?: "",
         price ?: BigDecimal.ZERO,
-        metaData?.get(0)?.value?.toString()?.toLongOrNull() ?: -1
```
Comment (hichamboushaba) [inline WooCommerce-Wear/src/main/java/com/woocommerce/android/wear/model/Refund.kt]: `JsonElement#toString()` will return String values wrapped in quotes, so we should use `valueAsString` instead, as it works well with all cases.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/model/OrderMapper.kt [hunk 2]
```diff
@@ -69,17 +70,9 @@ class OrderMapper @Inject constructor(
         )
     }
 
-    private fun List<WCMetaData>.getOrNull(key: String): String? = firstOrNull {
-        it.key == key
-    }.let {
-        it?.value?.toString()
-    }
+    private fun List<WCMetaData>.getOrNull(key: String): String? = get(key)?.valueAsString
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/model/OrderMapper.kt:73]: This is now using the extension `WCMetaData#get`

**Review by JorgeMucientes — APPROVED** [review]
Approving this one. I left my feedback on the FluxC side. Just needs a small review on an issue I found about json values. Once that is fixed this can be updated with the latest changeset and merged. 

Once again great job!

**Pull request merged**